src/measure_theory/prod.lean

```diff
@@ -0,0 +1,689 @@
+  (norm_of_nonneg $ integral_nonneg_of_ae $ eventually_of_forall $ λ y, (norm_nonneg _ : _)).symm
+
+lemma integrable.integral_prod_right [sigma_finite μ] ⦃f : α × β → E⦄
+  (hf : integrable f (μ.prod ν)) : integrable (λ y, ∫ x, f (x, y) ∂μ) ν :=
+hf.swap.integral_prod_left
+
+/-! ### The Bochner integral on a product -/
+
+variables [sigma_finite μ]
+
+lemma integral_prod_swap (f : α × β → E)
+  (hf : measurable f) : ∫ z, f z.swap ∂(ν.prod μ) = ∫ z, f z ∂(μ.prod ν) :=
+by rw [← integral_map measurable_swap hf, prod_swap]
+
+variables {E' : Type*} [measurable_space E'] [normed_group E'] [borel_space E'] [complete_space E']
+  [normed_space ℝ E'] [second_countable_topology E']
+
+/-! Some rules about the sum/difference of double integrals. They follow from `integral_add`, but
+  we separate them out as separate lemmas, because they involve quite some steps. -/
+lemma integral_fn_integral_add ⦃f g : α × β → E⦄
+  {F : E → E'} (hF : measurable F)
+  (hf : integrable f (μ.prod ν))
+  (hg : integrable g (μ.prod ν)) :
+  ∫ x, F (∫ y, f (x, y) + g (x, y) ∂ν) ∂μ = ∫ x, F (∫ y, f (x, y) ∂ν + ∫ y, g (x, y) ∂ν) ∂μ :=
+begin
+  refine integral_congr_ae
+    (hF.comp (hf.add hg).measurable.integral_prod_right')
+    (hF.comp (hf.measurable.integral_prod_right'.add hg.measurable.integral_prod_right')) _,
+  refine hg.prod_right_ae.mp _, refine hf.prod_right_ae.mp _,
+  apply eventually_of_forall, intros x h2f h2g, simp [integral_add h2f h2g]
+end
+
+lemma integral_fn_integral_sub ⦃f g : α × β → E⦄
+  {F : E → E'} (hF : measurable F)
+  (hf : integrable f (μ.prod ν))
+  (hg : integrable g (μ.prod ν)) :
+  ∫ x, F (∫ y, f (x, y) - g (x, y) ∂ν) ∂μ = ∫ x, F (∫ y, f (x, y) ∂ν - ∫ y, g (x, y) ∂ν) ∂μ :=
+begin
+  refine integral_congr_ae
+    (hF.comp (hf.sub hg).measurable.integral_prod_right')
+    (hF.comp (hf.measurable.integral_prod_right'.sub hg.measurable.integral_prod_right')) _,
+  refine hg.prod_right_ae.mp _, refine hf.prod_right_ae.mp _,
+  apply eventually_of_forall, intros x h2f h2g, simp [integral_sub h2f h2g]
+end
+
+lemma lintegral_fn_integral_sub ⦃f g : α × β → E⦄
+  (F : E → ennreal) (hf : integrable f (μ.prod ν)) (hg : integrable g (μ.prod ν)) :
+  ∫⁻ x, F (∫ y, f (x, y) - g (x, y) ∂ν) ∂μ = ∫⁻ x, F (∫ y, f (x, y) ∂ν - ∫ y, g (x, y) ∂ν) ∂μ :=
+begin
+  refine lintegral_congr_ae _,
+  refine hg.prod_right_ae.mp _, refine hf.prod_right_ae.mp _,
+  apply eventually_of_forall, intros x h2f h2g, simp [integral_sub h2f h2g]
+end
+
+lemma integral_integral_add ⦃f g : α × β → E⦄
+  (hf : integrable f (μ.prod ν))
+  (hg : integrable g (μ.prod ν)) :
+  ∫ x, ∫ y, f (x, y) + g (x, y) ∂ν ∂μ = ∫ x, ∫ y, f (x, y) ∂ν + ∫ y, g (x, y) ∂ν ∂μ :=
+integral_fn_integral_add measurable_id hf hg
+
+lemma integral_integral_add' ⦃f g : α × β → E⦄
+  (hf : integrable f (μ.prod ν))
+  (hg : integrable g (μ.prod ν)) :
+  ∫ x, ∫ y, (f + g) (x, y) ∂ν ∂μ = ∫ x, ∫ y, f (x, y) ∂ν + ∫ y, g (x, y) ∂ν ∂μ :=
+integral_integral_add hf hg
```

Comment:
What's the difference between primed and non-primed versions here? I failed to spot it, so probably it deserves a short docstring.

Reply:
Added docstrings for all double integration lemmas.